### ISSUE TITLE
RCTConvert.h yoga/Yoga.h file not found

### ISSUE THREAD

**Issue by adamgins** [facebook/react-native#11781]
I am logging this as a bug as it seems like a bunch of people have logged it as an issue on Stackoverflow ... but seems to be a bug http://stackoverflow.com/questions/41505616/yoga-yoga-h-header-file-not-found-after-react-native-integration-in-the-existing

Upgraded to latest react-native 0.4.0

Build project for iOS on Xcode and you get the error message: `/Users/adamginsburg/Documents/development/buzzy9/node_modules/react-native/React/Base/RCTConvert.h:13:9: 'yoga/Yoga.h' file not found`

* React Native version: 0.4.0
* Platform: iOS
* Operating System: The build issue happen on XCode 8

**Comment by adamgins** [comment]
Is this the reason? https://github.com/facebook/react-native/releases/tag/v0.40.0

**Comment by fgaleano** [comment]
I'm getting similar error messages. In my case:

- 'React/RCTConvert.h' file not found in RCTResizeMode.h
- 'React/RCTLog.h' file not found in RCTNetworkTask.m

I had like 5-6 more but  those went away after I deleted my `node_modules` folders and ran `npm install`. These 2 remained, though, and my app won't build.

What are we missing?

**Comment by julienfouilhe** [comment]
I am using react native in an already existing iOS app. Everything is working fine with `0.39.0`.

When I tried updating to 0.40.0, I get this error message when compiling, and I get this warning when running `pod install`:

```
[!] [Xcodeproj] Generated duplicate UUIDs
```

**Comment by emilsjolander** [comment]
Seems like the xcode project needs to be fixed in some way. Not really sure what could have broken this. @mmmulani Could you have a look?

**Comment by JesseARWhite** [comment]
This looks like it has something to do with Cocoapods integration, @julienfouilhe @emilsjolander can you confirm if this is coming from cocoapods or not?

**Comment by julienfouilhe** [comment]
I am using CocoaPods yes, but I don't know if it is responsible for the error or not. XCode is giving the error when trying to build the app.

**Comment by rh389** [comment]
I think this is the same issue we had in https://github.com/facebook/react-native/pull/9544, and occurs whenever `use_frameworks` is enabled - essentially a framework can't include its own internal headers using angle brackets.

Because FB needed to use angle brackets for its internal builds of CSSS Layout @alloy [proposed a compromise](https://github.com/facebook/react-native/pull/9544#issuecomment-242054698) which was removed in https://github.com/facebook/react-native/commit/7c91f894ba611d3e850c18008fab253cc652dad0 . Since the move to using angle brackets everywhere, the use of RN as a framework now looks pretty much beyond repair.

AFAIK it's still possible even in a swift project to disable `use_frameworks` and import RN via a bridging header. It should also be possible to find+replace imports in `post_install`, but that's really only a brittle hack if RN has dropped support for frameworks.

**Comment by julienfouilhe** [comment]
I'm using swift project and not using `use_frameworks!` would be painful, wouldn't it? Some of my dependencies are written in Swift.

**Comment by mmmulani** [comment]
cc @javache can you reapply for fix for the include header to yoga now?

**Comment by mmmulani** [comment]
hey everyone, I tried to repro this issue by creating a tiny project and following the Obj-C guide here: https://facebook.github.io/react-native/docs/integration-with-existing-apps.html
but I was not able to run into the Yoga build problem.

does anyone have a smaller repro case they could post?

**Comment by rh389** [comment]
I'll see if I can post a repro case tomorrow, I did reproduce the issue earlier - in the meantime have you tried enabling use_frameworks, even in Objective-C? I haven't yet been able to get that working as a minimal build in a new app at all without modifying RN (just importing a couple of essential react headers in Appdelegate.m).

**Comment by julienfouilhe** [comment]
Hi @mmmulani I also created a simple swift project, with all the files already in the repo, so all you have to do is launch the xcworkspace in xcode and try to build: https://github.com/julienfouilhe/test-yoga-file-not-found

**Comment by mmmulani** [comment]
@javache and I found a workaround to this problem and we'd like someone from the community to champion it and support it going forward.

we created a `yoga.podspec` at `node_modules/react-native/ReactCommon/yoga` with contents:

```
Pod::Spec.new do |s|
  s.name                = "yoga"
  s.source_files = '**/*.{c,h}'
  s.header_dir   = 'yoga'
  s.platform            = :ios, "8.0"
  s.author = ''
  s.license = 'a'
  s.homepage = 'a'
  s.source = { :path => '.' }
  s.summary = 'a'
  s.version = '0.1'
end
```

and then include it from the main Podfile with `pod 'yoga', :path => './node_modules/react-native/ReactCommon/yoga'`

you would have to make another one for jschelpers and possibly any other subspec we have that uses a different header search path.

but this seems like a bug in CocoaPods, it should let use create the React dependency and link all its subspecs statically but create the whole React dependency as a dylib. cc @alloy for any ideas here

**Comment by Aleksandern** [comment]
@mmmulani 
It looks like it doesn't work in my case 
I still have have an error

> RCTConvert.h:13:9: fatal error: 'yoga/Yoga.h' file not found
#import <yoga/Yoga.h>

But I had to change `pod 'yoga', :path => './node_modules/react-native/ReactCommon/yoga'` to `pod 'yoga', :path => '../node_modules/react-native/ReactCommon/yoga'`

    "react": "15.4.2",
    "react-native": "0.40.0",

**Comment by krzysztofp** [comment]
I had the same issue today.

When I searched in to the ../node-modules/react-native/ReactCommon/yoga directory, there was no podspec file, so I've manually created one and the install worked. But I really don't like the fact I need to alter some files in node-modules as this directory is not being tracked by my git repository

**Comment by krzysztofp** [comment]
Having the same issue, any progress on this?

**Comment by ghuh** [comment]
This solution probably won't work for everyone but when I commented out `use_frameworks!` in my pod file, the `yoga/Yoga.h file not found` error went away.

**Comment by yimingtang** [comment]
I'm making a [private pod](https://guides.cocoapods.org/making/private-cocoapods.html). The podspec file of version 0.40.0 fails CocoaPods lint command. It shows the `'yoga/Yoga.h' file not found` error from the compiler.

It's a sad story that I came accross a similar error in version 0.36.0 when I tried to integrate React Native into my existing app with CocoaPods. Now, the issue appears again.

NOTE: Commenting out `use_frameworks!` works. However, after `pod install`, thousands lines of warnings show on the console: `[!] [Xcodeproj] Generated duplicate UUIDs:`, even on a brand new Xcode project.

Did you seriously lint your podspec before a new release.

**Comment by Gagege** [comment]
Is there a solution where I don't have to comment out `use_frameworks!`? I'd really rather not, since I'm using React Native in an existing Swift project.

**Comment by renatomcamilio** [comment]
@Gagege I'm not sure I can call it a solution, but I'd say it's palliative at the very least: I've downgraded to `0.39.2`; Then you'll probably need the same fixes that I had to do for CSSLayout, since you also need `use_frameworks!`: https://github.com/facebook/react-native/issues/11272#issuecomment-268984452

**Comment by guysegal** [comment]
does this post install script repair the yoga problem? (by changing CSSLayout to Yoga)

**Comment by rh389** [comment]
After a bit of hacking around, this is now working for me in RN 0.40:

```ruby
# Uncomment this line to define a global platform for your project
platform :ios, '8.0'
# Uncomment this line if you're using Swift
use_frameworks!

target 'FrameworkDemo' do
    pod 'React', :subspecs => ['Core'], :path => './node_modules/react-native'
end

pre_install do |installer|
    `rm -rf node_modules/react-native/React/CSSLayout`
end

post_install do |installer|
    patch_subspecs(installer)
end

def patch_subspecs(installer)
    # patch source and header files
    `find node_modules/react-native/React node_modules/react-native/ReactCommon \\( -name *.h -o -name *.m -o -name *.mm \\) -print0 | xargs -0 sed -i '' -E 's:<(yoga|jschelpers|cxxreact)/(.*)>:"\\2":g'`

    # exclude duplicate symbols
    # https://github.com/facebook/react-native/issues/11502
    installer.pods_project.targets.each do |target|
        next unless target.name == 'React'

        source_files = target.source_build_phase.files
        uniqs = source_files.uniq { |f| f.file_ref.path }
        (source_files - uniqs).each { |dup| source_files.delete(dup) }
    end
end
```

Credit to @notjosh  at https://github.com/facebook/react-native/issues/11272#issuecomment-268984452 for the original.

**Comment by alloy** [comment]
@javache @mmmulani Just getting back on RN work from a hiatus. I fixed a podspec in the yoga repo yesterday, thinking it was somehow related to this, but am now thinking that it is not: https://github.com/facebook/yoga/pull/352

In [this other comment](https://github.com/CocoaPods/CocoaPods/issues/4605#issuecomment-275291022) @javache mentions “we've split the react-native podspec into multiple podspecs”, is that available somewhere?

Basically, what is the latest that I should be looking at and work from?

**Comment by JulianKingman** [comment]
I'm not using cocoapods and have the same issue. @adamgins, did you find a fix?

Related stack overflow question: http://stackoverflow.com/questions/31860715/rct-linking-manager-file-not-found

**Comment by adamgins** [comment]
@JulianKingman no, I downgraded to 0.39.2 and have been on vacation for the last week... so need to get back to this now.

**Comment by alloy** [comment]
@javache @mmmulani Got a PR for this up #12089

**Comment by adamgins** [comment]
@alloy thanks heaps... I have not tried this yet. Also wondering if anyone had seen any improvements with `0.4.0` https://github.com/facebook/react-native/releases/tag/v0.41.0-rc.0 ?

**Comment by debayanb** [comment]
Has there been any progress on this?

**Comment by GrahO23** [comment]
@rh389 thanks workaround works. Looking forward to the real fix :)

**Comment by quentinfasquel** [comment]
@GrahO23 it seems the workaround worked for me as long as I only use the 'Core' subspec, but when I import React in my Swift code, I encounter a compile issue with the C++ lib that's used: 'cstdint' file not found in JSBundleType.h Did you get that as well?

**Comment by GrahO23** [comment]
@quentinfasquel Yes sorry was a bit premature ;) , hadn't actually included the react view simply built with the pods in the xcode workscpace.

Getting the 'cstdint' file missing as well. Will let you know if I find a workaround

**Comment by GrahO23** [comment]
@adamgins No change with 0.41.0-rc.1

**Comment by adamgins** [comment]
@GrahO23 thanks for trying... ughhh.. was hoping for the magical 0.41

**Comment by the-simian** [comment]
I got the yoga error to go away by adding it to my header search paths

![screen shot 2017-02-03 at 9 05 47 pm](https://cloud.githubusercontent.com/assets/954596/22615197/427ce2d6-ea55-11e6-8183-d209cf3c11f4.png)

**Comment by admmasters** [comment]
So for our project - we added:
"${PODS_ROOT}/../node_modules/react-native/ReactCommon" to the header search path (recursive).
If that helps.

**Comment by quentinfasquel** [comment]
Well, it's closed but I still am not sure how to get things to compile.

**Comment by quentinfasquel** [comment]
It's closed but I still am not sure how to make it work.
I do have a Podfile like this
`  pod 'React', :path => './node_modules/react-native', :subspecs => [
    'Core',
    'RCTWebSocket',
  ]`

..and did an `npm install` which upgraded my version of React to 0.41.2

I still get a `cstdint file not found`
Thanks

**Comment by yimingtang** [comment]
The commit exists on master branch only at present. You'd better point your dependency to a branch or release which contains the fix. 

A quick fix is to change your `package.json`.

```
{
    ...,
    "dependencies": {
        "react-native": "facebook/react-native#master",
        ...,
    },
    ...
}
```

**Comment by AleF83** [comment]
Hey all,
After changing package.json to  `"react-native": "facebook/react-native#master"` I got 

`[!] Unable to find a specification for 'Yoga (= 1000.0.0.React)' depended upon by 'React/Core'`

when running `pod install`.

Thanks.

**Comment by alloy** [comment]
Be sure to add the Yoga pod to your Podfile as shown here https://github.com/facebook/react-native/pull/12089#issuecomment-277553904

**Comment by the-simian** [comment]
@alloy is that method preferable to the header file way I did it?

**Comment by alloy** [comment]
When using the new podspecs that are in master only right now, yeah. They do roughly the same thing, except it's all managed by CocoaPods which means in the future you wouldn't need to remember to remove/update those build settings.

**Comment by vevo-heyward-dai** [comment]
@alloy how do i test this out? change package.json to "react-native": "alloy/react-native#podspec" ?

**Comment by kmuralidharan91** [comment]
Guys I am using swift 3.0 and my **podfile** look likes this:

`# Uncomment the next line to define a global platform for your project
# platform :ios, '9.0'

target 'ReactNativeAnalysis' do

  # Comment the next line if you're not using Swift and don't want to use dynamic frameworks
  use_frameworks!
  # Pods
  pod 'Alamofire'
  pod 'SwiftyJSON'
  
  # Pods for ReactNativeAnalysis
  pod 'React', :path => './node_modules/react-native', :subspecs => [
  'Core',
  'RCTImage',
  'RCTNetwork',
  'RCTText',
  'RCTWebSocket',
  'yoga',
  ]
  
  target 'ReactNativeAnalysisTests' do
      inherit! :search_paths
      # Pods for testing
  end
  
  target 'ReactNativeAnalysisUITests' do
      inherit! :search_paths
      # Pods for testing
  end
  
end

pre_install do |installer|
    # Yoga exists in two places in the RN source tree. Delete the obsolete one to avoid redefinitions
    `rm -rf node_modules/react-native/React/CSSLayout`
end

post_install do |installer|
    patch_subspecs(installer)
end

def patch_subspecs(installer)
    # patch source and header files
    `find node_modules/react-native/React node_modules/react-native/ReactCommon \\( -name *.h -o -name *.m -o -name *.mm \\) -print0 | xargs -0 sed -i '' -E 's:<(yoga|jschelpers|cxxreact)/(.*)>:"\\2":g'`
    
    # exclude duplicate symbols
    # https://github.com/facebook/react-native/issues/11502
    installer.pods_project.targets.each do |target|
        next unless target.name == 'React'
        
        source_files = target.source_build_phase.files
        uniqs = source_files.uniq { |f| f.file_ref.path }
        (source_files - uniqs).each { |dup| source_files.delete(dup) }
    end
end`

I cant able to run this project because 'yoga/Yoga.h' is not found

**tried all the above fixes, its not working**

**and my package.json looks like below:**

`{
  "name": "ReactNativeAnalysis",
  "version": "1.0",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "bundle": "react-native bundle --platform ios --entry-file index.ios.js --bundle-output bundle/main.ios.jsbundle --assets-dest bundle/ --dev false",
    "postinstall": "find ./node_modules/react-native \\( -name *.h -o -name *.m \\) -print0 | xargs -0 sed -i '' -e 's:<CSSLayout/\\(.*\\)>:\"\\1\":g'"
  },
  "dependencies": {
    "react": "15.4.2",
    "react-native": "0.41.0",
    "react-native-tableview": "^1.7.0"
  },
  "devDependencies": {
    "babel-jest": "18.0.0",
    "babel-preset-react-native": "1.9.1",
    "jest": "18.1.0",
    "react-test-renderer": "15.4.2"
  },
  "jest": {
    "preset": "react-native"
  }
}
`

Guys can someone help me with the correct steps to solve this error.???

**Comment by julienfouilhe** [comment]
@kmuralidharan91 You don't need all the above fixes anymore, remove your patch function in your Podfile, remove the postinstall script in your package.json, just put the master branch of react-native in your dependencies and it should work.

**Comment by kmuralidharan91** [comment]
@julienfouilhe Worked like a charm. Thanks much brother 🥇

**Comment by julienfouilhe** [comment]
@alloy What release will this commit be available into? Can we expect it to be part of `0.42.0`?

**Comment by adamgins** [comment]
linking https://github.com/facebook/react-native/issues/12294

**Comment by yimingtang** [comment]
Good news: [v0.42.0-rc.3](https://github.com/facebook/react-native/releases/tag/v0.42.0-rc.3) is out!

**Comment by yuriyvl** [comment]
@julienfouilhe - I replicated the Podfile as well as package.json dependency.  Running `pod install` produces:

Analyzing dependencies
Fetching podspec for `React` from `./node_modules/react-native`
Fetching podspec for `Yoga` from `./node_modules/react-native/ReactCommon/yoga`
[!] No podspec found for `Yoga` in `./node_modules/react-native/ReactCommon/yoga`

Any idea?

**Comment by julienfouilhe** [comment]
Did you run `npm install` before that? Is your `node_modules` located in the same directory than your `Podfile` (if not, you must change `./node_modules` to `whatever/the/path/is/node_modules`?

**Comment by julienfouilhe** [comment]
@yuriyvl Did you run `npm install` before that? Is your `node_modules` located in the same directory than your `Podfile`? If not, you must change `./node_modules` to `whatever/the/path/is/node_modules`.

**Comment by yuriyvl** [comment]
@julienfouilhe Yup. Discovered right after I posted it :). Thank you very much.

**Comment by pbojinov** [comment]
After resolving the `RCTConvert.h yoga/Yoga.h file not found`, I'm now seeing the following now:

```
/Users/petar/dev/demo_app/node_modules/react-native/React/Executors/RCTJSCErrorHandling.mm:12:9: 'jschelpers/JavaScriptCore.h' file not found
```

Anyone come across this?

I'm on react-native version `0.42.0-rc.3` with Xcode 8.2.1.

**Comment by julienfouilhe** [comment]
That's funny, it's basically the same error. Are you using Pod? What's your Podfile like?

**Comment by anback** [comment]
Hi!

I have the same symtom as @pbojinov, here is my pod file for your reference. I will give it another go when 0.42 comes out. 


platform :ios, '8.0'
use_frameworks!
project 'Veckopengen.xcodeproj', 'Debug' => :debug, 'Staging' => :release, 'Beta' => :release, 'Production' => :release

target "Veckopengen" do
    pod 'Mixpanel', '<= 2.9.6'
    pod 'Firebase/Core'
    pod 'Firebase/Messaging'
    pod 'FBSDKCoreKit'
    pod 'FBSDKShareKit'
    pod 'FBSDKLoginKit'
    pod 'Intercom'

    react_path = '../node_modules/react-native'
    yoga_path = File.join(react_path, 'ReactCommon/yoga')

    pod 'React', :path => react_path, :subspecs => [
      'Core',
      'RCTText',
      'RCTImage',
      'RCTWebSocket', # needed for debugging
      'RCTNetwork',
      'RCTLinkingIOS'
    ]

    pod 'Yoga', :path => yoga_path
end

**Comment by alloy** [comment]
@pbojinov @anback To be sure, did you re-run `pod install`? If so, if you could share an example that exhibits the problem, that’d be helpful.

**Comment by pbojinov** [comment]
@julienfouilhe I'm using Pod. And I haven't changed any `#import` lines. The Pod file is below. 

After adding the separate `Yoga` pod, it ended up working.

```
platform :ios, '10.0'

target 'demoIOS' do

pod 'Firebase/Core'
pod 'Fabric'
pod 'Crashlytics'
pod 'GoogleSignIn'

pod 'React', :path => '../node_modules/react-native', :subspecs => [
    'Core',
    'RCTText',
    'RCTNetwork',
    'RCTImage',
    'RCTAnimation',
    'RCTWebSocket',
]

pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

end
```

And my folder structure looks like this:

```
├── README.md
├── app
│   ├── [REDACTED TO SAVE SPACE]
├── index.ios.js
├── ios
│   ├── Podfile
│   ├── Podfile.lock
│   ├── Pods
│   ├── build
│   ├── demoIOS
│   ├── demoIOS.xcodeproj
│   └── demoIOS.xcworkspace
├── node_modules
│   ├── [REDACTED TO SAVE SPACE]
└── package.json
```

**Comment by julienfouilhe** [comment]
@pbojinov Could you maybe push a basic example where the error occurs?

**Comment by pbojinov** [comment]
I'll try to see if I can reproduce it again.

**Comment by k7** [comment]
@julienfouilhe I have the example.
[SwiftReactNative.zip](https://github.com/facebook/react-native/files/795788/SwiftReactNative.zip)

the steps is 
```bash
npm install
pod install
xcodebuild -configuration Debug -scheme "SwiftReactNative" -workspace SwiftReactNative.xcworkspace | xcpretty
```

and it show
```bash
SwiftReactNative/node_modules/react-native/React/Modules/RCTDevMenu.mm:16:9: 'jschelpers/JavaScriptCore.h' file not found

#import <jschelpers/JavaScriptCore.h>
  ~~~~~~ ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by julienfouilhe** [comment]
@pbojinov @anback  Can you try updating cocoapods? I just tried with @k7 example and I don't have any problem compiling

**Comment by notjosh** [comment]
I can confirm updating CocoaPods to 1.2.0 fixed the `#import <jschelpers/JavaScriptCore.h>` issue on a project here, too.

**Comment by joshuafeldman** [comment]
I have CocoaPods version 1.2.0 and no issues running `pod install` but when I build the project I still see the issue with `#import <jschelpers/JavaScriptCore.h>`. I'm running the latest version of Xcode and the 0.42.0 release.

**Comment by joshuafeldman** [comment]
For those seeing the same issue as me to solve it I had to modify my own project's pod file to add `jschelpers` and `cxxreact` to the top of the `subspecs` of the `React` pod definition.

**Comment by joshuafeldman** [comment]
Looks like I spoke too soon. I'm getting farther in the compilation but now it fails with that same error inside of `RCTJSCErrorHandling.mm`

**Comment by rh389** [comment]
You definitely shouldn't need to add `cxxreact` and `jschelpers` to your Podfile explicitly, just `React` and `Yoga`. Could you post your `Podfile`? Have you tried the usual shaking out cobwebs (clean build in xcode, reinstall `node_modules`, `rm -rf Pods Podfile.lock && pod install`)?

**Comment by joshuafeldman** [comment]
@rh389 oh yeah I have cleaned like crazy. I retract that previous statement about that fixing it, it definitely did not.

The React section of my pod file looks like the following:

```
# React
pod 'React', :path => './Other Sources/ReactNative', :subspecs => [
  'Core',
  'RCTImage',
  'RCTNetwork',
  'RCTText',
  'RCTWebSocket',
  'ART',
  'RCTActionSheet',
  'RCTAdSupport',
  'RCTCameraRoll',
  'RCTGeolocation',
  'RCTPushNotification',
  'RCTSettings',
  'RCTVibration',
  'RCTLinkingIOS',
]
pod 'Yoga', :path => './Other Sources/ReactNative/ReactCommon/yoga'
```

Pod file configuration is simple
```
platform :ios, '8.0'
inhibit_all_warnings!
```

Definitely cannot get it to compile though. Stuck on the same file 
```
Other Sources/ReactNative/React/Modules/RCTDevMenu.mm:16:9: 'jschelpers/JavaScriptCore.h' file not found
```

Pretty much trying all kinds of ideas but any suggestions are welcomed

**Comment by joshuafeldman** [comment]
Also output for cocoa pods
```
$ pod --version
1.2.0
```

**Comment by rh389** [comment]
`Other Sources/ReactNative/...` is the only thing that looks suspicious there - is that a symlink? How did you install react native? You'd normally be pointing to `node_modules/react-native/..` installed using yarn or npm - I'm [pretty sure](https://github.com/facebook/react-native/search?utf8=%E2%9C%93&q=node_modules) the javascript side won't work unless you stick with that convention.

I'd double-check that you've actually got (at least) RN 0.42.0 at that path, and in particular that `Other Sources/ReactNative/React.podspec` looks like [this one](https://github.com/facebook/react-native/blob/0.42-stable/React.podspec) (except maybe the `version` string).

**Comment by joshuafeldman** [comment]
@rh389 we have been using this setup for over a year without issue. The issue only came up when upgrading to 0.42.0. All the correct files exist at the correct spot. `pod install` succeeds without any issues.

Essentially we checkin the contents of `react-native` to `Other Sources/ReactNative` so other iOS engineers not working with React do not have to install node to run the project.

**Comment by alloy** [comment]
@joshuafeldman Can you share a reduced example app that demonstrates the problem?

**Comment by joshuafeldman** [comment]
@alloy I will try and find some time this morning and wrap something up

**Comment by joshuafeldman** [comment]
@alloy @rh389 I created a really fast application that displays the compilation error. Hopefully it is just something simple I am doing wrong :)

https://github.com/joshuafeldman/react-native-missing-file

**Comment by ambientlight** [comment]
@joshuafeldman, @alloy:
Run into the same thing. 

Went into the Pods.xcodeproj within Xcode. Then opened  React target **Build Settings**. Looked into **Header Search Paths**, $(PODS_TARGET_SRCROOT) environment variable doesn't get resolved. (cocoapods issue?)
So replacing it with the full absolute path to **react-native** essentially resolves this issue at the project builds fine.

Steps to reproduce (cocoapods version 1.1.1):
1. Create a new folder with **ios** and **js** folders inside.
2. Create a new swift iOS project and place it into **ios**.
3. In **js** create a package.json with following content:
```js
{
  "name": "test",
  "version": "1.0.0",
  "private": true,
  "description": "Test",
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start"
  },
  "dependencies": {
    "react": "~15.4.2",
    "react-native": "~0.42.0"
  }
}
```
4. npm install (in **js**)
5. in **iOS** create Podfile with this content:
```
platform :ios, '8.0'
inhibit_all_warnings!
use_frameworks!

target 'Test'
pod 'Yoga', :path => '../js/node_modules/react-native/ReactCommon/yoga'
pod 'React', :path => '../js/node_modules/react-native', :subspecs => [
  'Core',
  'RCTImage',
  'RCTNetwork',
  'RCTText',
  'RCTWebSocket',
]
```
6. pod install (in **iOS**) goes fine.
7. open workspace -> build -> same issue as @joshuafeldman

**Comment by alloy** [comment]
@ambientlight You need CocoaPods version >= 1.2.0. Can you double-check if that fixes it for you?

**Comment by alloy** [comment]
@joshuafeldman Awesome, thanks! Will take a look later today.

**Comment by ambientlight** [comment]
@alloy yes, I can confirm that with cocoaPods **1.2.0** it builds fine out of the box. thanks.

**Comment by Elik86** [comment]
@joshuafeldman, @alloy: 
same problem here.

- update pod to 1.2.0

- run "pod update" 

- clean & build

worked for me  

@ambientlight:

i used in my
Build Settings ==> Header Search Paths 

"$(SRCROOT)/Pods/Headers/Public/" - for the environment variable

**Comment by joshuafeldman** [comment]
@alloy the project I linked to was setup using cocoapods version 1.2.0 and exhibits the issue. I wonder what the difference is.

**Comment by alloy** [comment]
@joshuafeldman I gave it a spin, the issue is the space in the `./Other Sources` path. This is definitely a bug in the podspec.

I’ll make a PR for this, but in the meantime you can work around it by either:

* Removing the whitespace from `Other Sources`
* Patch `Other Sources/ReactNative/React.podspec` to replace `"$(PODS_TARGET_SRCROOT)/ReactCommon"` with `"\"$(PODS_TARGET_SRCROOT)/ReactCommon\""`

**Comment by joshuafeldman** [comment]
@alloy there it is 🙌

Appreciate the insight.

**Comment by QuantumInformation** [comment]
My builds are failing like so:

```
The following build commands failed:
        Analyze /Users/nikos/WebstormProjects/TTracker/node_modules/react-native/ReactCommon/yoga/yoga/YGNodeList.c
        Analyze /Users/nikos/WebstormProjects/TTracker/node_modules/react-native/ReactCommon/yoga/yoga/Yoga.c
(2 failures)```

I'll need to look at this pods things

```
"dependencies": {
    "react": "16.0.0-alpha.6",
    "react-native": "0.44.0-rc.0",
    "react-navigation": "^1.0.0-beta.9"
  },
  "devDependencies": {
    "babel-jest": "19.0.0",
    "babel-preset-react-native": "1.9.1",
    "jest": "19.0.2",
    "react-test-renderer": "16.0.0-alpha.6"
  },
```

**Comment by gotoAndBliss** [comment]
This error occurs again on a fresh install of `react-native: 0.44.0`

**Comment by QuantumInformation** [comment]
@gotoAndBliss try deleting your ios build folder

**Comment by cliren** [comment]
I am running into the following error and have setup a simple example for anyone to advise what's going wrong here: https://github.com/cliren/RNIntegration. 

Using cocopods version: 1.1.1 (can't upgrade due to exiting app's dependency). Appreciate any help!

```
RNIntegration/js/node_modules/react-native/React/Modules/RCTDevSettings.mm:16:9: fatal error: 'jschelpers/JavaScriptCore.h' file not found
#import <jschelpers/JavaScriptCore.h>
        ^
1 error generated.

```

**Comment by tmaly1980** [comment]
I had a similar issue with yoga.c, and fixed it by running `react-native link`

**Comment by cliren** [comment]
@tmaly1980 I tried your suggestion on https://github.com/cliren/RNIntegration but still ran into the same issue. I am using the below dependencies, may I know what versions are you using?

```
"react": "16.0.0-alpha.12",
"react-native": "0.45.1"
```

**Comment by tmaly1980** [comment]
I have to stick with 0.42.3 and 15.4.1 because of some packages that refuse to support alpha code (react 16.0.0)

**Comment by cliren** [comment]
@tmaly1980 I tried 0.42.3 and 15.4.1  but still get the same error. It looks very fragile unless I am missing something basic.  Would you mind checking if I am missing anything here: https://github.com/cliren/RNIntegration? 

```
react-native-cli: 2.0.1
react-native: 0.42.3
```

**Comment by tmaly1980** [comment]
Sorry, I honestly don’t know, I am using ignite as a boilerplate/scaffolding and they have a command line script out that generates my project. Not to sound like a sales person, it saves me a ton of time with having a project structure in place and generating components, containers, redux/sagas, etc. I tried upgrading to 16.0.0.alpha-2 but that breaks some of the testing suite I’m relying upon (as the core API has changed but THOSE test library developers refuse to support alpha software).

**Comment by samthui** [comment]
After upgrading from RN 0.42.3 to 0.43.3, I got similar issue with RCTDevMenu, though I'm sure these file exist.
My project also uses CocoaPods for iOS project. After re-running `pod install` and restart MAC, it works normally. 
I noticed that in terminal console of `pod install`, it logged:

> Installing React 0.43.3 (was 0.42.3)
Installing Yoga 0.43.3.React (was 0.42.3.React)

(Thanks to that I know that CocoaPods need to re-config RN dependency by running `pod install`, previously I thought that CocoaPods only needs to be configured once and use path to its dependencies all afterwards.)

**Comment by rodriguesbreno** [comment]
For the ones having this issue with RN 0.47 or above,
I've discovered that it appear in a project I'm working after change the disk to case sensitive.
CocoaPods seems name the Yoga project as <Yoga/Yoga.h> not <yoga/Yoga.h> 

Revert the disk fixed (at least locally) the issue.
